### DISCUSSION TITLE
Developing a WinUI 3 Project Convertor

### DISCUSSION THREAD

**Discussion by MarkBrainfire** [unoplatform/uno#13395] [General]
Would anyone be interested in working with me to create a utility that would convert WinUI 3 Applications and Libraries into Uno Platform Applications and Libraries?  The motivation for this is that it's faster and better to develop a project as a WinUI 3 application that is ready to run on billions of Window devices.  Then a utility could be run that would convert this WinUI 3 application into a multi-headed Uno Platform Application enabling the application to run on non-Window devices and the Web.  The utility could first detect any controls or features not supported by Uno Platform and suggest alternatives.  Once everything was compatible, it would generate the necessary files for an Uno Platform application.

WinUI 3 Project -> Uno Platform Project -> Everywhere  
... aligns Uno Platform with the future of Microsoft's desktop development.

Mark

**Comment by jeromelaban** [comment]
[Our documentation](https://platform.uno/docs/articles/howto-migrate-existing-code.html) also contains steps that could be used to build such a tool.
